core/domain/topic_services.py

```diff
@@ -201,6 +201,24 @@ def get_topic_by_id(topic_id, strict=True, version=None):
             return None
 
 
+def get_topic_by_name(topic_name):
+    """Returns a domain object representing a topic.
+
+    Args:
+        topic_name: str. The name of the topic.
+
+    Returns:
+        Topic or None. The domain object representing a topic with the
+        given id, or None if it does not exist.
+    """
+    topic_model = topic_models.TopicModel.get_by_name(topic_name)[0]
+    if topic_model:
+        topic = get_topic_from_model(topic_model)
+        return topic
+    else:
```

Comment:
else is not required

Reply:
Okay, I dealt with empty differently. Fixed.